Stock.playground/Contents.swift

```diff
@@ -3,116 +3,147 @@ import Foundation
 
 struct StockGrant
 {
-    struct GrantDate
-    {
-        private(set) var foundationDate: NSDate
-
-        init (month: Int, day: Int, year: Int)
-        {
-            let signing = NSDateComponents.init()
-            signing.calendar = NSCalendar.currentCalendar()
-            signing.month = month
-            signing.day = day
-            signing.year = year
-            self.foundationDate = signing.date!
-        }
-    }
-
-
     let shares: UInt
     let strikePrice: Double
-    let grantDate: GrantDate
+    let grantDate: Date
     let cliffMonths: UInt
     let endMonths: UInt
 
-
-    // RSUs
-    init(shares: UInt, grantDate: GrantDate, cliffMonths: UInt, endMonths: UInt)
+    // RSU grants will use default value for strikePrice, NSO grants will set a strike price.
+    init(shares: UInt, grantDate: Date, cliffMonths: UInt, endMonths: UInt, strikePrice: Double = 0)
     {
-        self.strikePrice = 0
+        self.strikePrice = strikePrice
         self.shares = shares
         self.grantDate = grantDate
         self.cliffMonths = cliffMonths
         self.endMonths = endMonths
     }
 
-    // NSOs
-    init (shares: UInt, grantDate: GrantDate, cliffMonths: UInt, endMonths: UInt, strikePrice: Double)
+    // Bonus grants: vest instantly
+    init(shares: UInt, grantDate: Date, strikePrice: Double = 0)
     {
-        self.strikePrice = strikePrice
         self.shares = shares
+        self.strikePrice = strikePrice
         self.grantDate = grantDate
-        self.cliffMonths = cliffMonths
-        self.endMonths = endMonths
+        cliffMonths = 0
+        endMonths = 0
     }
 
-    func valueAfter(months: Int, atPrice: Double) -> Double
+    func valueAt(_ price: Double, monthsElapsed months: UInt) -> Double
```

Comment:
I really like the comment about using a type alias. It helps with semantics and would ease the conversion to an actual money type later.